### PR TITLE
Feature tmc#99 for loop

### PR DESCRIPTION
**Pull request by rphh**

### CODE REVIEW

File: sdl2promela/translator.py [hunk 1]
```diff
@@ -212,7 +215,10 @@ def __is_local_variable(context: Context, variable: str):
             for parameter in procedure.parameters:
                 if parameter.name == variable:
                     return True
-    return False
+    if variable.lower() in context.sdl_model.variables:
+        return False
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:219]: Why?

File: sdl2promela/translator.py [hunk 2]
```diff
@@ -837,6 +851,73 @@ def __generate_for_over_a_numeric_range(
...
+def __generate_for_over_sequenceof(
+    context: Context,
+    transition: sdlmodel.Transition,
+    task: sdlmodel.ForLoopTask,
+    statements: List[promelamodel.Statement],
+):
+    block_builder = BlockBuilder(promelamodel.BlockType.BLOCK)
+
+    range = typing.cast(sdlmodel.ForEachLoopRange, task.range)
+
```
Comment (Lurkerpas) [inline sdl2promela/translator.py:863]: Why so many newlines?

File: sdl2promela/translator.py (hunk 1 above)
Comment (rphh) [reply sdl2promela/translator.py:219]: The iterator variable in `for` is a local variable, to detect this I added this check.

File: tests/unit/translator/resources/for_each.pml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      global_state.taskwithloop.state = Taskwithloop_States_wait;
+      transition_id = -1;
+    ::(transition_id == 1)->
+      MyInteger_assign_value(global_state.taskwithloop.result_param, 0);
+      {
+        MyInteger each;
+        int i;
+        for(i : 0 .. (global_state.taskwithloop.variable_param.length - 1))
+        {
+          each = global_state.taskwithloop.variable_param.data[i];
```
[the remark this answers is not in the document]
Comment (rphh) [reply tests/unit/translator/resources/for_each.pml]: indeed, fixed

[the remark this answers is not in the document]
Comment (rphh) [reply tests/unit/translator/resources/for_each.pml]: Also, the translator.py

File: sdl2promela/translator.py (hunk 2 above)
Comment (rphh) [reply sdl2promela/translator.py]: changed.

File: sdl2promela/translator.py [hunk 4]
```diff
@@ -837,6 +851,73 @@ def __generate_for_over_a_numeric_range(
...
+    basic_type = find_basic_type(context.sdl_model.source.dataview, range.variableType)
+
+    block_builder.withStatement(
+        VariableDeclarationBuilder(
+            task.iteratorName.variableName, range.type.CName
+        ).build()
+    )
+
+    block_builder.withStatement(VariableDeclarationBuilder("i", "int").build())
```
[the remark this answers is not in the document]
Comment (rphh) [reply sdl2promela/translator.py]: Solved, I've added a test for this case (nested loop). I've checked the generated code manually - it returns valid value.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**